### ISSUE TITLE
LiveChartsCore.SkiaSharpView.Avalonia 2.0.0-beta.800-11.0.0-rc1.1 freezes

### ISSUE THREAD

**Issue by Developer-Alexander** [beto-rodriguez/LiveCharts2#1076]
**Describe the bug**
Avalonia app freezes, when you zoom very deeply into chart. It seems to happen only when only a few points are visible or when the visible interval on the x axis is very small.

Additional Info:

- The source collection of the chart is updated very frequently in the background.
- The source collection is used as SyncContext.
- Point type is DateTimePoint.
- ZoomMode is ZoomAndPanMode.X.
- LiveChartsCore.SkiaSharpView.Avalonia: 2.0.0-beta.800-11.0.0-rc1.1
- Avalonia: 11.0.0-rc1.1

**To Reproduce**
See attached minimal app.
[ChartTest_freeze.zip](https://github.com/beto-rodriguez/LiveCharts2/files/11785145/ChartTest_freeze.zip)

**Expected behavior**
The app should not freeze.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows

**Comment by beto-rodriguez** [comment]
Thanks for the project.

The app is not freezing for me, it gets sluggish when you have around 20k points.

This is the expected behavior, you can remove the points that you do not longer need, you could also compute your data before adding it to reduce the number of points, or you could wait for the official release of the high-performance package, I mean in general the performance seems nice, about 20k points, we can zoom in/out but to get more performance you need to run a virtualization algorithm, for more info see: https://github.com/beto-rodriguez/LiveCharts2/releases/tag/v2.0.0-beta.350

Am I missing something?

**Comment by beto-rodriguez** [comment]
Never mind, I got the issue.

This happens because when you zoom so deep then the double type has not enough precision to handle the current range.

I think I can fix this internally, while this is fixed you can set a limit in the axis zoom, for example in this case I force the zoom to stop at 1ms:

```c#
        XAxes = new Axis[]
        {
            new Axis
            {
                MinZoomDelta = TimeSpan.FromMilliseconds(1).Ticks,
            }
        };
```

**Comment by beto-rodriguez** [comment]
For now there is just a warning, in the next versions of the library, we will throw when we detect this issue, A possible solution is to set the `MinZoomDelta ` to a significant value depending on the case.

In cases where setting where `MinZoomDelta` is not enough (because the chart has values that the double precision can't handle) you could try to scale your chart using mappers similar to what we do in the [Logarithms example](https://lvcharts.com/docs/WPF/2.0.0-beta.800/samples.axes.logarithmic).

@Developer-Alexander feel free to reply or reopen the issue if the issue persists.

**Comment by beto-rodriguez** [comment]
The referenced commit fixes this, I think it is a smart way to prevent this issue, in the next version of the library, setting a min limit will not be a requirement.

**Comment by guilhermeolisi** [comment]
I’m using LiveChart2 (version 2.0.0-rc2) in an Avalonia 11 (version 11.0.9) project. My control includes 3 CartesianCharts, and I frequently need to zoom and pan these charts. However, I’m encountering a recurring exception related to “an excessive number of separators,” which causes my application to crash. This issue does not occur with deep zooming, but rather when I only pan the chart or apply a moderate zoom.

The exception arises within the GetPossibleMaxLabelSize() method of the LiveChartsCore.CoreAxis class. After analyzing the source code, I attempted to circumvent the issue by setting values for MinZoomDelta and MinStep, but the problem persists. I also tried controlling the MaxLimit and MinLimit of the axis, yet the exception continues to be thrown.

I have captured the values of all axis properties and non-public members of each axis in my project at the time the exception occurred (using the debug tools of Visual Studio 2022). The data is included in the attached file of this message. Unfortunately, I am unable to pinpoint the source of the problem; the property and variable values do not seem to justify the exception. It’s possible that the instance when I retrieved the axes values differs from the moment the exception is thrown.

Any assistance would be greatly appreciated. I am very fond of LiveChart2; it’s an incredible tool for data visualization, but this exception is causing my app to crash frequently.

[LiveChart2 Axis exception.txt](https://github.com/beto-rodriguez/LiveCharts2/files/15131495/LiveChart2.Axis.exception.txt)
